# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/bees/hotfix-openat2.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Backport upstream 14c82dce8a7e and f1558d64a58c: match the openat2
prototype declared by glibc >= 2.43.

--- bees-0.11.orig/include/crucible/openat2.h	2025-06-19 03:17:48.000000000 +0200
+++ bees-0.11/include/crucible/openat2.h	2026-09-27 18:40:50.628564088 +0200
@@ -45,7 +45,7 @@ struct open_how {
 extern "C" {
 
 /// Weak symbol to support libc with no syscall wrapper
-int openat2(int dirfd, const char *pathname, struct open_how *how, size_t size) throw();
+int openat2(int dirfd, const char *pathname, const struct open_how *how, size_t size);
 
 };
 
--- bees-0.11.orig/lib/openat2.cc	2025-06-19 03:17:48.000000000 +0200
+++ bees-0.11/lib/openat2.cc	2026-09-27 18:40:50.628596410 +0200
@@ -26,8 +26,7 @@ extern "C" {
 
 int
 __attribute__((weak))
-openat2(int const dirfd, const char *const pathname, struct open_how *const how, size_t const size)
-throw()
+openat2(int const dirfd, const char *const pathname, const struct open_how *const how, size_t const size)
 {
 #ifdef SYS_openat2
 	return syscall(SYS_openat2, dirfd, pathname, how, size);
